types/dns2/dns2-tests.ts

```diff
@@ -139,6 +139,49 @@ serverOnAddress.listen({
     },
 });
 
+const serverOnAddressWithUdpType = DNS.createServer({
+    udp: true,
+    handle: (request, send, rinfo) => {
+        const response = Packet.createResponseFromRequest(request);
+        const [question] = request.questions;
+        const { name } = question;
+        response.answers.push({
+            name,
+            type: Packet.TYPE.A,
+            class: Packet.CLASS.IN,
+            ttl: 300,
+            address: "8.8.8.8",
+        });
+        response.answers.push({
+            name,
+            type: Packet.TYPE.CNAME,
+            class: Packet.CLASS.IN,
+            ttl: 300,
+            domain: "another-name.example.com",
+        });
+        response.answers.push({
+            name,
+            type: Packet.TYPE.TXT,
+            class: Packet.CLASS.IN,
+            ttl: 60,
+            data: "dnslink=/ipfs/123abc",
+        });
+        send(response);
+    },
+});
+
+serverOnAddressWithUdpType.on("request", (request, response, rinfo) => {
+    console.log(request.header.id, request.questions[0]);
+});
+
+serverOnAddressWithUdpType.listen({
+    udp: {
+        port: 5355,
+        address: "0.0.0.0",
+        type: "udp4",
```

Comment:
I know the README says this will work, but looking at the code I think it's wrong. Specifically because:

* This `listen` method calls `server.listen` on the given server type ([here](https://github.com/song940/node-dns/blob/b2fdf66a1987e2597ad1b1f0948d0ac5c4acbfe2/server/dns.js#L64-L74)) in one of two ways:
  * If there's a `port` argument, it explicitly uses just the port & address from the options and throws the rest away, so in `{ port, address, type }` examples like this the type is definitely not used
  * In other cases, it passes the options directly to `server.listen` (i.e. `UdpDnsServer.listen`) but the `type` argument is only used in the constructor ([here](https://github.com/song940/node-dns/blob/b2fdf66a1987e2597ad1b1f0948d0ac5c4acbfe2/server/udp.js#L9-L20)) which has already been called before that point (during `createServer`).

I've just done a very rough test:

* Clone dns2 locally
* Add `console.log('UDP type', type);` before [this `super(type)`](https://github.com/song940/node-dns/blob/b2fdf66a1987e2597ad1b1f0948d0ac5c4acbfe2/server/udp.js#L15) call where it's used
* Run `require('.').createServer({ udp: true }).listen({ port: 1234, type: 'udp6' })`
* It prints `UDP type udp4` (it ignores this type argument completely).

Maybe I'm missing something, but to me it looks like the README example is wrong and actually this option should be passed to `createServer`. For example:

```
> require('.').createServer({ udp: { type: 'udp6' } }).listen({ port: 1234 })
UDP type udp6
```

That does do what you're looking for. Does that make sense? Given that, I think we should probably close this PR and update the README in dns2 instead.